### PR TITLE
Update sdk version and check for subdomain uniqueness

### PR DESCRIPTION
**Pull request by r-delgadillo**
* Updated to latest version of azure-mgmt-iotcentral python sdk
* Added check for subdomain uniqueness.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by r-delgadillo** [comment]
Add @emgarten @zzcodework @amrenbin @cle5eland @vavilla

File: src/command_modules/azure-cli-iotcentral/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+0.1.4
++++++
+* Use latest azure-mgmt-iotcentral version.
```
Comment (emgarten) [inline src/command_modules/azure-cli-iotcentral/HISTORY.rst]: You can leave this off, a cli user doesn't need to know this

File: src/command_modules/azure-cli-iotcentral/azure/cli/command_modules/iotcentral/custom.py [hunk 2]
```diff
@@ -66,6 +67,16 @@ def _check_name_availability(client, app_name):
...
 
 
+def _check_subdomain_availability(client, app_subdomain):
+    """Search the current subscription for an app with the given subdomain.
+    :param object client: IoTCentralClient
+    :param str app_subdomain: App subdomain to search for
+    """
+    subdomain_availability = client.apps.check_subdomain_availability(app_subdomain)
+    if subdomain_availability is not None and not subdomain_availability.name_available:
+        raise CLIError(subdomain_availability.message)
```
Comment (emgarten) [inline src/command_modules/azure-cli-iotcentral/azure/cli/command_modules/iotcentral/custom.py]: what does RP send for this? Will the user be able to tell the difference between a conflict on resource name and a conflict on subdomain?

Comment (r-delgadillo) [reply src/command_modules/azure-cli-iotcentral/azure/cli/command_modules/iotcentral/custom.py]: Discussed offline.

File: src/command_modules/azure-cli-iotcentral/HISTORY.rst (hunk 1 above)
Comment (r-delgadillo) [reply src/command_modules/azure-cli-iotcentral/HISTORY.rst]: removed

File: src/command_modules/azure-cli-iotcentral/azure/cli/command_modules/iotcentral/custom.py [hunk 3]
```diff
@@ -66,6 +67,16 @@ def _check_name_availability(client, app_name):
         raise CLIError(name_availability.message)
 
 
+def _check_subdomain_availability(client, app_subdomain):
+    """Search the current subscription for an app with the given subdomain.
+    :param object client: IoTCentralClient
+    :param str app_subdomain: App subdomain to search for
+    """
+    subdomain_availability = client.apps.check_subdomain_availability(app_subdomain)
+    if subdomain_availability is not None and not subdomain_availability.name_available:
```
Comment (williexu) [inline src/command_modules/azure-cli-iotcentral/azure/cli/command_modules/iotcentral/custom.py]: When would `subdomain_availability` be None?

Comment (r-delgadillo) [reply src/command_modules/azure-cli-iotcentral/azure/cli/command_modules/iotcentral/custom.py]: I was following the same pattern as the existing method for `check_name_availability`. I guess this shouldn't ever be `None` I can remove this if you'd like.

Comment (r-delgadillo) [reply src/command_modules/azure-cli-iotcentral/azure/cli/command_modules/iotcentral/custom.py]: applied 😃 .

**Comment by r-delgadillo** [comment]
@tjprescott For `checkNameAvailability` and `checkSubdomainAvailability` our rp-service will return a message stating if the given name/subdomain is available. If the name/subdomain is taken than it will return a message stating it's taken. 

We do these checks because on our `appCreate` api we do not check for uniquness of the name or subdomain, we assume at this point they are unique. When users create an application using our UI or Azure Portal we do this check prior to creating an application as well. Also, if we do not do these checks the user may confuse a `create` with an `update` event because the response is the same from our rp-service.

**Comment by r-delgadillo** [comment]
@tjprescott Discussed offline, submitted an issue to make IoTCentral create command idempotent. #7791

**Pull request merged**